src/CSET/operators/mesoscale.py

```diff
@@ -0,0 +1,123 @@
+# © Crown copyright, Met Office (2022-2024) and CSET contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A module containing different diagnostics for mesoscales.
+
+The diagnostics here are applicable at mesoscales and apply generally
+rather than for specific aspects of mesoscale meteorology (e.g. convection).
+For specific aspects, the user is referred to other modules available in
+CSET.
+
+"""
+
+import logging
+
+import iris
+import numpy as np
+from scipy.ndimage import gaussian_filter, uniform_filter
+
+from CSET.operators._utils import get_cube_yxcoordname
+
+
+def spatial_perturbation_field(
+    original_field: iris.cube.Cube,
+    Gaussian_filter: bool = True,
+    filter_scale: int = 40,
+) -> iris.cube.Cube:
+    """Calculate a spatial perturbation field.
+
+    Parameters
+    ----------
+    original_field: iris.cube.Cube
+        Iris cube containing data to smooth, supporting multiple dimensions
+        (at least two spatial dimensions must be supplied, i.e. 2D).
+    Gaussian_filter: boolean
+        Switch to determine if a Gaussian filter is applied.
+        If set to True a Gaussian filter is applied; if set to False
+        a Uniform filter is applied.
+        Default is True.
+    filter_scale: int
+        Scale at which to define the filter in grid boxes. If the
+        filter is a Gaussian convolution this value represents the
+        halfwidth of the Gaussian kernel.
+        Default is 40 grid points.
+
+    Returns
+    -------
+    An iris cube of the spatial perturbation field.
+
+    Return type
+    -----------
+    iris.cube.Cube
+
+    Notes
+    -----
+    In mesoscale meteorology the perturbation field is more important than the
+    balanced flows for process understanding. This function is designed
+    to create spatial perturbation fields based on smoothing with a Gaussian
+    kernel or a uniform kernel.
+
+    The kernels are defined by the filter_scale, which for mesoscale
+    perturbations should be between an approximate cloud separation distance,
+    of 30 km, and synoptic scale variations (1000 km). In practice any
```

Comment:
Given that we know(?) what the filter scale should be, should we calculate it inside the operator rather than have it user defined?

Reply:
The whole point is we do not know, there will be some applications where we want to apply different size filters, thus it should remain user defined.